src/odm_sharing/tools/share.py

```diff
@@ -209,28 +202,39 @@ def share(
                     org_data[table] = sh.get_data(con, tq)
 
             # one excel file per org
-            excel_file = get_excel_writer(in_name, debug, org, outdir, outfmt)
+            excel_fn = gen_filename(in_name, org, '', 'xlsx')
+            excel_path = os.path.join(outdir, excel_fn)
+            excel_file = None
+            if not debug and outfmt == OutFmt.EXCEL:
+                excel_file = pd.ExcelWriter(excel_path, engine='openpyxl')
+                logging.info('writing ' + excel_fn)
             try:
                 for table, data in org_data.items():
                     if outfmt == OutFmt.CSV:
                         filename = gen_filename(in_name, org, table, 'csv')
-                        logging.info('writing ' + filename)
                         path = os.path.join(outdir, filename)
+                        logging.info('writing ' + filename)
                         data.to_csv(path, index=False)
+                        result.append(path)
                     elif outfmt == OutFmt.EXCEL:
                         logging.info(f'- {qt(table)}')
                         data.to_excel(excel_file, sheet_name=table,
                                       index=False)
                     else:
                         assert False, f'format {outfmt} not impl'
+                if outfmt == OutFmt.EXCEL:
```

Comment:
yes, since we don't want to add a result file before it's fully written, but that was the wrong place since it's not finished before after it's closed. I pushed a new commit with the fix.